implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -99,57 +160,148 @@ public void reportHealth(OutputStream out, SmallRyeHealth health) {
     }
 
     public SmallRyeHealth getHealth() {
-        return getHealth(healthChecks, livenessChecks, readinessChecks);
+        return getHealth(smallRyeHealthUni, HEALTH, LIVENESS, READINESS);
     }
 
     public SmallRyeHealth getLiveness() {
-        return getHealth(livenessChecks);
+        return getHealth(smallRyeLivenessUni, LIVENESS);
     }
 
     public SmallRyeHealth getReadiness() {
-        return getHealth(readinessChecks);
+        return getHealth(smallRyeReadinessUni, READINESS);
     }
 
     public SmallRyeHealth getHealthGroup(String groupName) {
-        return getHealth(allHealthChecks.select(HealthGroup.Literal.of(groupName)));
+        return getHealthGroupAsync(groupName).await().atMost(Duration.ofSeconds(timeoutSeconds));
     }
 
-    @Inject
-    BeanManager beanManager;
-
     public SmallRyeHealth getHealthGroups() {
-        Iterator<Bean<?>> iterator = beanManager.getBeans(HealthCheck.class, Any.Literal.INSTANCE).iterator();
+        return getHealthGroupsAsync().await().atMost(Duration.ofSeconds(timeoutSeconds));
+    }
+
+    @Experimental("Asynchronous Health Check procedures")
+    public Uni<SmallRyeHealth> getHealthAsync() {
+        return getHealthAsync(smallRyeHealthUni, HEALTH, LIVENESS, READINESS);
+    }
 
-        List<HealthCheck> groupHealthChecks = new ArrayList<>();
+    @Experimental("Asynchronous Health Check procedures")
+    public Uni<SmallRyeHealth> getLivenessAsync() {
+        return getHealthAsync(smallRyeLivenessUni, LIVENESS);
```

Comment:
Are `getLivenessAsync` and `getReadinessAsync` still used (outside of this class)?

Reply:
They are a public experimental API. I would expect that they can be integrated in user applications.